docs/multi-stage-query/concepts.md

```diff
@@ -102,6 +104,19 @@ issues](./known-issues.md#select) page.
 
 For more information about the syntax, see [REPLACE](./reference.md#replace).
 
+When deciding whether to use REPLACE or INSERT, keep in mind that REPLACE can perform dimension-based segment pruning but INSERT cannot. To use dimension-based segment pruning, your query needs to meet the following requirements:
+
+- Segments were generated by a REPLACE statement, not an INSERT statement.
```

Comment:
I don't think this section needs to be reiterated here as this info is already captured later in the "Clustering" section. I think just linking to that section is enough.